store/schedule_test.go

```diff
@@ -203,35 +328,77 @@ func TestValidateSchedule(t *testing.T) {
 			CronExpression: "*/5 * * * *",
 		}
 
-		errs := s.ValidateSchedule()
+		errs := s.ValidateSchedule(a, conf)
 		if len(errs) == 0 {
 			t.Fatal("expected errors, got none")
 		}
 	})
 
 	t.Run("valid recurring schedule", func(t *testing.T) {
+		conf := conf2.AppLevelConfiguration{
+			FutureScheduleCreationPeriod: 7,
+			FiredScheduleRetentionPeriod: 1,
+			PayloadSize:                  1024,
+			HttpRetries:                  1,
+			HttpTimeout:                  1000,
+		}
+
+		a := App{
+			AppId:      "appId",
+			Partitions: 0,
+			Active:     false,
+			Configuration: Configuration{
+				FutureScheduleCreationPeriod: 7,
+				FiredScheduleRetentionPeriod: 1,
+				PayloadSize:                  1024,
+				HttpRetries:                  1,
+				HttpTimeout:                  1000,
+			},
+		}
+
 		s := &Schedule{
 			AppId:          "test-app-id",
 			Payload:        "test-payload",
 			Callback:       &MockCallback{Field: "success"},
 			CronExpression: "* * * * *",
 		}
 
-		errs := s.ValidateSchedule()
+		errs := s.ValidateSchedule(a, conf)
 		if len(errs) != 0 {
 			t.Fatalf("expected no errors, got %v", errs)
 		}
 	})
 
 	t.Run("valid one time schedule", func(t *testing.T) {
+		conf := conf2.AppLevelConfiguration{
+			FutureScheduleCreationPeriod: 7,
+			FiredScheduleRetentionPeriod: 1,
+			PayloadSize:                  1024,
+			HttpRetries:                  1,
+			HttpTimeout:                  1000,
+		}
+
+		a := App{
+			AppId:      "appId",
+			Partitions: 0,
+			Active:     false,
+			Configuration: Configuration{
+				FutureScheduleCreationPeriod: 7,
+				FiredScheduleRetentionPeriod: 1,
+				PayloadSize:                  1024,
+				HttpRetries:                  1,
+				HttpTimeout:                  1000,
+			},
+		}
+
 		s := &Schedule{
 			AppId:        "test-app-id",
 			Payload:      "test-payload",
 			Callback:     &MockCallback{Field: "success"},
 			ScheduleTime: time.Now().Unix() + 100,
 		}
 
-		errs := s.ValidateSchedule()
+		errs := s.ValidateSchedule(a, conf)
```

Comment:
ValidateSchedule method is not accepting any method argument, check if passing a and conf as argument is correct.

Reply:
The modified method accepts both app and configs.